datafusion/physical-expr/src/expressions/like.rs

```diff
@@ -0,0 +1,365 @@
+                array,
+                scalar.clone(),
+                nilike,
+                &DataType::Boolean
+            ),
+        };
+        Ok(scalar_result)
+    }
+
+    fn evaluate_array_array(
+        &self,
+        left: Arc<dyn Array>,
+        right: Arc<dyn Array>,
+    ) -> Result<ArrayRef> {
+        match (self.negated, self.case_insensitive) {
+            (false, false) => binary_string_array_op!(left, right, like),
+            (true, false) => binary_string_array_op!(left, right, nlike),
+            (false, true) => binary_string_array_op!(left, right, ilike),
+            (true, true) => binary_string_array_op!(left, right, nilike),
+        }
+    }
+}
+
+/// Create a like expression whose arguments are correctly coerced.
+/// This function errors if it is not possible to coerce the arguments
+/// to computational types supported by the operator.
+pub fn like(
+    negated: bool,
+    case_insensitive: bool,
+    expr: Arc<dyn PhysicalExpr>,
+    pattern: Arc<dyn PhysicalExpr>,
+    input_schema: &Schema,
+) -> Result<Arc<dyn PhysicalExpr>> {
+    let expr_type = &expr.data_type(input_schema)?;
+    let pattern_type = &pattern.data_type(input_schema)?;
+    if !expr_type.eq(pattern_type) {
+        return Err(DataFusionError::Internal(format!(
+            "The type of {expr_type} AND {pattern_type} of like physical should be same"
+        )));
+    }
+    Ok(Arc::new(LikeExpr::new(
+        negated,
+        case_insensitive,
+        expr,
+        pattern,
+    )))
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::expressions::col;
+    use arrow::array::BooleanArray;
+    use arrow_schema::Field;
+    use datafusion_common::cast::as_boolean_array;
+
+    macro_rules! test_like {
+        ($A_VEC:expr, $B_VEC:expr, $VEC:expr, $NEGATED:expr, $CASE_INSENSITIVE:expr,) => {{
+            let schema = Schema::new(vec![
+                Field::new("a", DataType::Utf8, false),
+                Field::new("b", DataType::Utf8, false),
+            ]);
+            let a = StringArray::from($A_VEC);
+            let b = StringArray::from($B_VEC);
+
+            let expression = like(
+                $NEGATED,
+                $CASE_INSENSITIVE,
+                col("a", &schema)?,
+                col("b", &schema)?,
+                &schema,
+            )?;
+            let batch = RecordBatch::try_new(
+                Arc::new(schema.clone()),
+                vec![Arc::new(a), Arc::new(b)],
+            )?;
+
+            // compute
+            let result = expression.evaluate(&batch)?.into_array(batch.num_rows());
+            let result =
+                as_boolean_array(&result).expect("failed to downcast to BooleanArray");
+            let expected = &BooleanArray::from($VEC);
+            assert_eq!(expected, result);
+        }};
+    }
+
+    #[test]
+    fn like_op() -> Result<()> {
+        test_like!(
+            vec!["hello world", "world"],
```

Comment:
It would be great to add `NULL` to these cases as well -- e.g.

```suggestion
            vec![Some("hello world"), None, Some("world")],
```

To verify that the operator correctly handles nulls